api-client/src/test/java/com/xing/api/CallSpecTest.java

```diff
@@ -489,6 +493,60 @@ public void specHandlesSuccessResponse() throws Exception {
         assertSuccessResponse(response, new TestMsg("success", 42));
     }
 
+    @Test
+    public void specFailsWhenHostnameVerifierCannotVerify() {
+        // Custom url is required to have `https`
+        httpUrl = server.url("https://www.google.com");
+        mockApi = new XingApi.Builder().custom()
+                .apiEndpoint(httpUrl)
+                .build();
+
+        HostnameVerifier noGoogleVerifier = new HostnameVerifier() {
+            @Override
+            public boolean verify(String host, SSLSession sslSession) {
+                return !host.contains("google");
```

Comment:
I suggest to avoid `!host.contains..` and `host.contains` logic. Instead make it more explicitly.
`return false;` and `return true;` for next test. In that case names could be `alwaysFailsVerifier` and `alwaysSucceedVerifier`.

Reply:
Thanks for the tests @aballano, that looks great. I agree with @MyDogTom that making it explicit would make it easier to get the point of the test.

Theoretically, only a single test that makes sure that the given verifier was used during the request would be sufficient for the code changes, isn't it? Something like this where `didUseHostnameVerifier` would be asserted to equal `true` at the end:

```java
@Test
    public void canOverrideHostnameVerifier() {
        boolean didUseHostnameVerifier = false;
        HostnameVerifier noGoogleVerifier = new HostnameVerifier() {
            @Override
            public boolean verify(String host, SSLSession sslSession) {
                didUseHostnameVerifier = true;
                return true;
```

What do you think?

Reply:
I agree with @MyDogTom in making the test more explicit but I disagree in the verification of usage. The reason is that In my approach we test behaviour (that is not really under our control), aka we test that what we meant to happen, happens; while with this other we would test that their logic is simply calling the verifier, but we cannot be sure that is really doing anything.

Reply:
@MyDogTom @aballano I applied the requested changes, please have a second look. Thank you.